test/test_bdf_reactors.py

```diff
@@ -0,0 +1,314 @@
+"""
+Cantera example problem: reactor2.py
+
+Two reactors connected with a piston, with heat loss to the environment
+
+This script simulates the following situation. A closed cylinder with volume 2
+m^3 is divided into two equal parts by a massless piston that moves with speed
+proportional to the pressure difference between the two sides.  It is
+initially held in place in the middle. One side is filled with 1000 K argon at
+20 atm, and the other with a combustible 500 K methane/air mixture at 0.1 atm
+(phi = 1.1). At t = 0 the piston is released and begins to move due to the
+large pressure difference, compressing and heating the methane/air mixture,
+which eventually explodes. At the same time, the argon cools as it expands.
+The piston is adiabatic, but some heat is lost through the outer cylinder
+walls to the environment.
+
+Note that this simulation, being zero-dimensional, takes no account of shock
+wave propagation. It is somewhat artifical, but nevertheless instructive.
+"""
+
+import numpy as np
+import pytest
+
+
+def test_vs_scipy():
+    pytest.importorskip("scipy")
+    pytest.importorskip("cantera")
+    pytest.importorskip("pyrometheus")
+    pytest.importorskip("jax")
```

Comment:
We should make it so that this actually runs in CI.

Reply:
Yeah, I can take care of this. I know I had to rip out a Scipy `importorskip` in a previous PR, so I can look there to remember how I did it.

Reply:
I'll help. You'll need to set up a Conda-based CI, which I think I can probably do more quickly.

Reply:
#20

Reply:
08017c2 and 01f6f8a should do the trick.

Reply:
Maybe a38620c  and 7e0d1b9?